src/main/java/uk/gov/hmcts/darts/event/service/impl/EventServiceImpl.java

```diff
@@ -19,11 +23,26 @@ public class EventServiceImpl implements EventService {
     private final EventMapper eventMapper;
     private final EventIdValidator eventIdValidator;
     private final EventRepository eventRepository;
+    private final DataAnonymisationService dataAnonymisationService;
 
     @Override
     public AdminGetEventForIdResponseResult adminGetEventById(Integer eventId) {
         eventIdValidator.validate(eventId);
         Optional<EventEntity> eventEntityOptional = eventRepository.findById(eventId);
         return eventMapper.mapToAdminGetEventsResponseForId(eventEntityOptional);
     }
+
+    @Override
+    @Transactional
+    public void adminObfuscateEveByIds(List<Integer> eveIds) {
+        eveIds.stream()
+            .map(this::getEventEntityById)
+            .distinct()
+            .forEach(dataAnonymisationService::anonymizeEvent);
+    }
+
+    EventEntity getEventEntityById(Integer eveId) {
+        return eventRepository.findById(eveId)
+            .orElseThrow(() -> new DartsApiException(DartsApiException.DartsApiErrorCommon.NOT_FOUND));
```

Comment:
dont we want an operation specific error code?

Reply:
there seems to be already an event specific error code we can reuse, see [here](https://github.com/hmcts/darts-api/blob/a67c3b3234dc52445c5a34e4ef4e78808e2821b9/src/main/java/uk/gov/hmcts/darts/event/validation/EventIdValidator.java#L24).
(even thought the enum name is quite bad :) )

Reply:
For non specific errors such as 404 I think using common errors across the bored make sense instead of entity specific ones such as EventApiError.NOT_FOUND, CourtApiError.NOT_FOUND, etc

Reply:
If we want to define a common 404 error then thats fine, but the response body should always contain a problem with an associated app error code. The app error code provides the unique code around the specific operation failure. Thats the pattern we have adopted across the system so far.

Reply:
Error codes do still get added using the prefix COMMON

So not found would be COMMON_NOT_FOUND

Reply:
@jackmaloney  What are your thoughts?

Reply:
Sorry only just seen this, and now it's been merged. 
I think we should be following the pattern that we have used elsewhere instead of deviating. If we think we need to change to a different pattern then fine, but as Mario says it needs agreement with the project. Why can't we use the same error code that Mario referenced above @Ben-Edwards-cgi ?

Reply:
The primary purpose of the DARTS-specific error codes are to inform the caller that a specific failure has occured, such that they can handle it in a specific way. The caller's error-handling code is keyed on the specific DARTS error code.

So it's down to what the API design specifies - if we want the client to be able to disambiguate between an event not found error and (say) a courthouse not found error, then it makes sense to have a specific error code/enum for the event not found scenario. If the client doesn't care, then a generic "not found" is fine. There is no specific pattern for this that I'm aware of, it's use-case/design specific.

However, there is a pattern for how we define our errors in code - If a generic error is suitable, then I suggest it be defined in the existing `src/main/java/uk/gov/hmcts/darts/common/exception/CommonApiError.java` rather than the new inner class within `src/main/java/uk/gov/hmcts/darts/common/exception/DartsApiException.java`. The error code and human-readable message must be defined as enums in the openapi yaml - `src/main/resources/openapi/common.yaml`, and those generated enums used within the enum definition within CommonApiError. The idea is that our error codes and messages are stable and form part of the API contract, therefore are defined in an openapi spec. The convention is the error code (errorTypeNumeric in the error enum) takes the form `FEATURE_XXX`, where XXX is a numerical value unique to the feature. e.g `COMMON_100`, `COMMON_110` etc, and we should stick to that.

I would say if we're going for a generic error then I'd encourage use of the `public DartsApiException(DartsApiError error, String detail)` constructor, and provide some description in the `detail`. The use of a common error code won't allow a client to key their error handling, but the `detail` will at least give devs additional info should we need to troubleshoot the error in the future. E.g:

```
new DartsApiException(DartsApiException.DartsApiErrorCommon.NOT_FOUND, "The provided event id does not correspond to an existing event"));
```

Reply:
I did not realise we had a CommonApiError.java :)

I'm happy to either move the existing CommonApiError into the inner class (as these errors are exclusively used within that exception so we don't really need a standard class for it) or to remove them from the inner and stick the items in the common.

I also like the idea of having a changeable detail section as it gives us the ability to use common errors while allowing for more detailed logging but I think it would be good to have a "default" value for when it does not matter